### ISSUE TITLE
Dependency clash while trying to implement datadog sdk v1.7.0

### ISSUE THREAD

**Issue by hs-kdhillon** [DataDog/dd-sdk-android#464]
**Describe what happened**
After including datadog SDK in the project, there are dependency clashes that exist for `androidx` namespace, namely for the following dependencies:
> cv: Compile version, rv: Runtime version
```
androidx.fragment:fragment (cv:1.1.0, rv:1.2.4)
androidx.activity:activity (cv:1.0.0, rv:1.1.0)
androidx.core:core (cv:1.1.0, rv:1.3.1)
androidx.lifecycle:lifecycle-service (cv:2.0.0, rv:2.1.0)
```
To resolve this, manual gradle dependency resolution was done as follow:
```
subprojects {
        project.configurations.all {
            resolutionStrategy.eachDependency { details ->
                if (details.requested.group == 'androidx.activity'
                  && details.requested.name.contains('activity') ) {
                    details.useVersion "1.0.0"
                }
            }
            resolutionStrategy.eachDependency { details ->
                if (details.requested.group == 'androidx.fragment'
                  && details.requested.name.contains('fragment') ) {
                    details.useVersion "1.1.0"
                }
            }
            resolutionStrategy.eachDependency { details ->
                if (details.requested.group == 'androidx.lifecycle'
                  && details.requested.name.contains('lifecycle-service') ) {
                    details.useVersion "2.0.0"
                }
            }
            resolutionStrategy.eachDependency { details ->
                if (details.requested.group == 'androidx.core'
                  && details.requested.name.contains('core') ) {
                    details.useVersion "1.1.0"
                }
            }
        }
    }
```
Running `.gradlew :app:installDebug --scan` leads to a crash with the following crashlog:
```
> Task :app:transformClassesWithDexBuilderForDebug
com.android.builder.dexing.DexArchiveBuilderException: com.android.builder.dexing.DexArchiveBuilderException: Failed to process <PACKAGE_NAME>/app/build/intermediates/transforms/FirebasePerformancePlugin/debug/145
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at java.util.concurrent.ForkJoinTask.getThrowableException(ForkJoinTask.java:593)
	at java.util.concurrent.ForkJoinTask.reportException(ForkJoinTask.java:677)
	at java.util.concurrent.ForkJoinTask.join(ForkJoinTask.java:720)
	at com.android.ide.common.internal.WaitableExecutor.waitForTasksWithQuickFail(WaitableExecutor.java:146)
	at com.android.build.gradle.internal.transforms.DexArchiveBuilderTransform.transform(DexArchiveBuilderTransform.java:405)
	at com.android.build.gradle.internal.pipeline.TransformTask$2.call(TransformTask.java:239)
	at com.android.build.gradle.internal.pipeline.TransformTask$2.call(TransformTask.java:235)
	at com.android.builder.profile.ThreadRecorder.record(ThreadRecorder.java:102)
	at com.android.build.gradle.internal.pipeline.TransformTask.transform(TransformTask.java:230)
	at sun.reflect.GeneratedMethodAccessor992.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:73)
	at org.gradle.api.internal.project.taskfactory.IncrementalTaskAction.doExecute(IncrementalTaskAction.java:50)
	at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:39)
	at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:26)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$1.run(ExecuteActionsTaskExecuter.java:131)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:300)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:292)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:174)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:90)
	at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:120)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:99)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:77)
	at org.gradle.api.internal.tasks.execution.OutputDirectoryCreatingTaskExecuter.execute(OutputDirectoryCreatingTaskExecuter.java:51)
	at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:59)
	at org.gradle.api.internal.tasks.execution.ResolveTaskOutputCachingStateExecuter.execute(ResolveTaskOutputCachingStateExecuter.java:54)
	at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:59)
	at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:101)
	at org.gradle.api.internal.tasks.execution.FinalizeInputFilePropertiesTaskExecuter.execute(FinalizeInputFilePropertiesTaskExecuter.java:44)
	at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:91)
	at org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter.execute(ResolveTaskArtifactStateTaskExecuter.java:62)
	at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:59)
	at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:54)
	at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
	at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:34)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.run(EventFiringTaskExecuter.java:51)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:300)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:292)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:174)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:90)
	at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:46)
	at org.gradle.execution.taskgraph.LocalTaskInfoExecutor.execute(LocalTaskInfoExecutor.java:42)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareWorkItemExecutor.execute(DefaultTaskExecutionGraph.java:273)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareWorkItemExecutor.execute(DefaultTaskExecutionGraph.java:258)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:135)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:130)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker.execute(DefaultTaskPlanExecutor.java:200)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker.executeWithWork(DefaultTaskPlanExecutor.java:191)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker.run(DefaultTaskPlanExecutor.java:130)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
	at java.lang.Thread.run(Thread.java:748)
Caused by: com.android.builder.dexing.DexArchiveBuilderException: Failed to process <PACKAGE_NAME>/app/build/intermediates/transforms/FirebasePerformancePlugin/debug/145
	at com.android.build.gradle.internal.transforms.DexArchiveBuilderTransform.launchProcessing(DexArchiveBuilderTransform.java:900)
	at com.android.build.gradle.internal.transforms.DexArchiveBuilderTransform.lambda$convertToDexArchive$6(DexArchiveBuilderTransform.java:825)
	at java.util.concurrent.ForkJoinTask$AdaptedCallable.exec(ForkJoinTask.java:1424)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
Caused by: com.android.builder.dexing.DexArchiveBuilderException: Error while dexing.
	at com.android.builder.dexing.D8DexArchiveBuilder.getExceptionToRethrow(D8DexArchiveBuilder.java:124)
	at com.android.builder.dexing.D8DexArchiveBuilder.convert(D8DexArchiveBuilder.java:101)
	at com.android.build.gradle.internal.transforms.DexArchiveBuilderTransform.launchProcessing(DexArchiveBuilderTransform.java:895)
	... 6 more
Caused by: com.android.tools.r8.CompilationFailedException: Compilation failed to complete
	at com.android.tools.r8.utils.ExceptionUtils.withCompilationHandler(ExceptionUtils.java:65)
	at com.android.tools.r8.utils.ExceptionUtils.withD8CompilationHandler(ExceptionUtils.java:43)
	at com.android.tools.r8.D8.run(D8.java:90)
	at com.android.builder.dexing.D8DexArchiveBuilder.convert(D8DexArchiveBuilder.java:99)
	... 7 more
Caused by: com.android.tools.r8.utils.AbortException: Error: <PACKAGE_NAME>/app/build/intermediates/transforms/FirebasePerformancePlugin/debug/145/META-INF/versions/9/module-info.class, Illegal class file: Class module-info is missing a super type. Class file version 53.
	at com.android.tools.r8.utils.Reporter.failIfPendingErrors(Reporter.java:116)
	at com.android.tools.r8.utils.Reporter.fatalError(Reporter.java:74)
	at com.android.tools.r8.utils.ExceptionUtils.withCompilationHandler(ExceptionUtils.java:59)
	... 10 more


> Task :app:transformClassesWithDexBuilderForDebug FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:transformClassesWithDexBuilderForDebug'.
> com.android.build.api.transform.TransformException: com.android.builder.dexing.DexArchiveBuilderException: com.android.builder.dexing.DexArchiveBuilderException: Failed to process <PACKAGE_NAME>/app/build/intermediates/transforms/FirebasePerformancePlugin/debug/145

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 28s
289 actionable tasks: 9 executed, 280 up-to-date

```
Could this be an issue between `FirebasePerformacePlugin` and `datadog SDK`?

**Steps to reproduce the issue:**

1. Added `datadog SDK` maven repo in project's `build.gradle.kts`.
2. Added `datadog dependency` `implementation "com.datadoghq:dd-sdk-android:1.7.0"` to app's `build.gradle.kts`.
3. Setup datadog in `Application` Class `onCreate()` as follow:
> CLIENT_TOKEN, ENV_NAME, and APP_ID redacted.
```
DatadogConfig datadogConfig = new DatadogConfig.Builder(
      "CLIENT_TOKEN",
      "<ENV_NAME>",
      "<APP_ID>"
    )
      .trackInteractions()
      .setTracesEnabled(true)
      .setRumEnabled(true)
      .useViewTrackingStrategy(new FragmentViewTrackingStrategy(true))
      .build();

    Datadog.initialize(this, TrackingConsent.GRANTED, datadogConfig);

    RumMonitor datadogRumMonitor = new RumMonitor.Builder()
      .sampleRumSessions(100.0f)
      .build();
    GlobalRum.registerIfAbsent(datadogRumMonitor);
```
4. Build the project.

**Describe what you expected:**
Expected the project to build without any dependency conflicts.

**Additional context**
 - Android OS version: 10
 - Device Model: Samsung galaxy S10
 - Datadog SDK version: 1.7.0
 - retrofit: 2.7.1 (not using `DatadogInterceptor()`)
 - Proguard configuration: default

 - Android Gradle plugin: 3.2.0
 - Kotlin Gradle plugin: 1.3.61
 - Google Services plugin: 3.1.1
 - Firebase Crashlytics plugin: 2.3.0
 - Firebase Performance plugin: 1.3.1

PS: I am trying to upgrade the dependencies to their latest versions, especially the ones with `androidx` and `firebase`. I will keep the issue posted with the findings.

**Comment by xgouchet** [comment]
Hello @hs-kdhillon, can you share with us the `dependencies` block in your gradle file. Without that we won't be able to reproduce this on our end. 

Regarding the issue itself, when there is a conflict in transitive dependencies (one library depends on androidx.fragment:fragment `1.1.0`, the other on `1.2.4`), Gradle will automatically bump every one to use the latest.

Although this doesn't look like what your stacktrace is saying. Googling around, I found this [stackoverflow issue](https://stackoverflow.com/questions/58369285/illegal-class-file-class-module-info-is-missing-a-super-type-class-file-versio) that might be related to your actual issue, and is linked to a bug in AGP was fixed in version `3.6` (you're using AGP `3.2` right ?)

**Comment by hs-kdhillon** [comment]
Here are the `dependencies` for project's `build.gradle`:
```
dependencies {
        classpath BuildPlugins.kotlinGradlePlugin
        classpath BuildPlugins.androidGradlePlugin
        classpath BuildPlugins.googleServicesPlugin
        classpath BuildPlugins.realmPlugin
        classpath BuildPlugins.firebaseCrashlyticsPlugin
        classpath BuildPlugins.firebasePerformancePlugin
    }
```
Here are the `dependencies` for app's `build.gradle`:
```
dependencies {
    implementation fileTree(dir: 'libs', include: ['*.jar'])
    api project(':common')
    implementation(Libs.androidSupportLibraryV7)
    api(Libs.dagger)
    annotationProcessor(Libs.daggerCompiler)
    kapt(Libs.daggerCompiler)
    api(Libs.retrofit)
    api(Libs.retrofitGsonConverter)
    api(Libs.retrofitRxJavaAdapter)
    api(Libs.okhttp3LoggingInterceptor)
    api(Libs.icepick)
    annotationProcessor(Libs.icepickProcessor)
    kapt(Libs.icepickProcessor)
    api(Libs.timber)
    api(Libs.googlePlayServicesAnalytics)
    api(Libs.googlePlayServicesBase)
    api(Libs.googlePlayServicesGoogleCloudMessaging)
    api(Libs.googlePlayServicesLocation)
    api(Libs.googlePlayServicesMaps)
    api(Libs.googlePlayServicesAuth)
    api(Libs.googleMapsUtil)
    api(Libs.firebaseRemoteConfig)
    api(Libs.jodaTime)
    api(Libs.mixpanel)
    api(Libs.firebaseAnalytics)
    api(Libs.firebaseNewCrashlytics)
    testImplementation(Libs.leakCanaryNoOp)
    debugImplementation(Libs.leakCanary)
    releaseImplementation(Libs.leakCanaryNoOp)
    api(Libs.processPhoenixx)
    api(Libs.branch) { exclude module: 'answers-shim' }
    api(Libs.uxCam)
    api(Libs.onRadar) { exclude module: 'support-v4' exclude module: 'play-services-location' }
    api(Libs.calligraphy)
    api(Libs.viewpump)
    annotationProcessor(Libs.dataBindingAnnotationProcessor)
    api(Libs.fancyButtons)
    api(Libs.gifDrawable)
    api(Libs.picassoTransformations)
    api(Libs.percentDimensions)
    api(Libs.glide)
    api(Libs.realmAdapter)
    annotationProcessor(Libs.parceler)
    kapt(Libs.parceler)
    api(Libs.swipeLayout)
    api(Libs.grantlandAutoFitTextView)
    api(Libs.roundedImageView)
    api(Libs.androidSupportLibraryCardView)
    api(Libs.androidSupportConstraintLayout)
    api(Libs.circleImageView)
    api(Libs.kotlinCoroutinesCore)
    api(Libs.kotlinCoroutinesAndroid)
    api(Libs.androidxLifecycleViewModel)
    implementation(Libs.androidxLifecycleOnly)
    implementation(Libs.androidxLifecycleLivedata)
    implementation(Libs.materialMenu)
    implementation(Libs.nekocodeBadge)
    implementation "com.datadoghq:dd-sdk-android:1.7.0"
}
```
Here is the `Dependencies.kt` file under `BuildSrc`:
```
object BuildPlugins {
    object Versions {
        const val kotlinGradlePlugin = "1.3.61"
        const val androidGradlePlugin = "3.2.0"
        const val googleServicesPlugin = "3.1.1"
        const val realmPlugin = "4.3.2"
        const val firebaseCrashlyticsPlugin = "2.3.0"
        const val firebasePerformancePlugin = "1.3.1"
    }

    const val kotlinGradlePlugin = "org.jetbrains.kotlin:kotlin-gradle-plugin:${Versions.kotlinGradlePlugin}"
    const val androidGradlePlugin = "com.android.tools.build:gradle:${Versions.androidGradlePlugin}"
    const val googleServicesPlugin = "com.google.gms:google-services:${Versions.googleServicesPlugin}"
    const val realmPlugin = "io.realm:realm-gradle-plugin:${Versions.realmPlugin}"
    const val firebasePerformancePlugin = "com.google.firebase:perf-plugin:${Versions.firebasePerformancePlugin}"
    const val firebaseCrashlyticsPlugin = "com.google.firebase:firebase-crashlytics-gradle:${Versions.firebaseCrashlyticsPlugin}"
}

object ProjectVersions {
    const val kotlinVersion = "1.3.61"
    const val androidSupportLibrary = "1.0.0"
    const val googlePlayServices = "17.0.0"
    const val buildTools = "26.0.2"
    const val minSdk = 21
    const val compileSdk = 29
    const val versionCode = 465
    const val versionName = "4.1.75"
}

object Libs {
    const val crashlytics = "com.crashlytics.sdk.android:crashlytics:2.6.7@aar"
    const val crashlyticsAnswers = "com.crashlytics.sdk.android:answers:1.3.12@aar"
    const val androidSupportLibraryV7 = "androidx.appcompat:appcompat:${ProjectVersions.androidSupportLibrary}"
    const val androidSupportLibraryV4 = "androidx.legacy:legacy-support-v4:1.0.0"
    const val androidSupportLibraryV4Compat = "androidx.core:core:${ProjectVersions.androidSupportLibrary}"
    const val androidxLegacySupportLibraryV13 = "androidx.legacy:legacy-support-v13:${ProjectVersions.androidSupportLibrary}"
    const val androidSupportLibraryDesign = "com.google.android.material:material:1.1.0"
    const val androidSupportLibraryRecyclerView = "androidx.recyclerview:recyclerview:${ProjectVersions.androidSupportLibrary}"
    const val androidSupportConstraintLayout = "androidx.constraintlayout:constraintlayout:1.1.3"
    const val androidSupportTransition = "androidx.transition:transition:${ProjectVersions.androidSupportLibrary}"
    const val androidSupportTestRunner = "androidx.test:runner:1.0.1"
    const val androidSupportLibraryCardView = "androidx.cardview:cardview:1.0.0"
    const val androidSdkCrashlytics = "com.crashlytics.sdk.android:crashlytics:2.10.1"
    const val firebaseNewCrashlytics = "com.google.firebase:firebase-crashlytics:17.0.0"
    const val firebaseCore = "com.google.firebase:firebase-core:${ProjectVersions.googlePlayServices}"
    const val firebaseRemoteConfig = "com.google.firebase:firebase-config:${ProjectVersions.googlePlayServices}"
    const val firebaseAnalytics = "com.google.firebase:firebase-analytics:${ProjectVersions.googlePlayServices}"
    const val firebasePerformance = "com.google.firebase:firebase-perf:${ProjectVersions.googlePlayServices}"
    const val googlePlayServicesBase = "com.google.android.gms:play-services-base:${ProjectVersions.googlePlayServices}"
    const val googlePlayServicesGoogleCloudMessaging = "com.google.android.gms:play-services-gcm:${ProjectVersions.googlePlayServices}"
    const val googlePlayServicesLocation = "com.google.android.gms:play-services-location:${ProjectVersions.googlePlayServices}"
    const val googlePlayServicesMaps = "com.google.android.gms:play-services-maps:${ProjectVersions.googlePlayServices}"
    const val googlePlayServicesAuth = "com.google.android.gms:play-services-auth:${ProjectVersions.googlePlayServices}"
    const val googlePlayServicesAnalytics = "com.google.android.gms:play-services-analytics:${ProjectVersions.googlePlayServices}"
    const val googleMapsUtil = "com.google.maps.android:android-maps-utils:0.5"
    const val materialLibrary = "com.google.android.material:material:1.1.0"
    const val icepick = "frankiesardo:icepick:3.2.0"
    const val icepickProcessor = "frankiesardo:icepick-processor:3.2.0"
    const val retrofit = "com.squareup.retrofit2:retrofit:2.7.1"
    const val retrofitGsonConverter = "com.squareup.retrofit2:converter-gson:2.7.1"
    const val retrofitRxJavaAdapter = "com.squareup.retrofit2:adapter-rxjava2:2.3.0"
    const val okhttp3LoggingInterceptor = "com.squareup.okhttp3:logging-interceptor:3.9.0"
    const val picasso = "com.squareup.picasso:picasso:2.5.2"
    const val nekocodeBadge = "com.github.nekocode:Badge:1.2"
    const val timber = "com.jakewharton.timber:timber:4.7.1"
    const val jodaTime = "joda-time:joda-time:2.9.4"
    const val mixpanel = "com.mixpanel.android:mixpanel-android:4.6.4"
    const val grantlandAutoFitTextView = "me.grantland:autofittextview:0.2.1"
    const val kotlinCoroutinesCore = "org.jetbrains.kotlinx:kotlinx-coroutines-core:1.3.3"
    const val kotlinCoroutinesAndroid = "org.jetbrains.kotlinx:kotlinx-coroutines-android:1.3.3"
    const val androidxLifecycleViewModel = "androidx.lifecycle:lifecycle-viewmodel-ktx:2.2.0"
    const val androidxLifecycleLivedata = "androidx.lifecycle:lifecycle-livedata-ktx:2.2.0"
    const val androidxLifecycleOnly = "androidx.lifecycle:lifecycle-runtime-ktx:2.2.0"
    const val androidLifecycleOnly = "androidx.lifecycle:lifecycle-runtime:2.0.0"
    const val parceler = "org.parceler:parceler:1.1.5"
    const val parcelerApi = "org.parceler:parceler-api:1.1.5"
    const val freeBuilder = "org.inferred:freebuilder:1.14.1"
    const val javaAnnotations = "javax.annotation:jsr250-api:1.0"
    const val junit = "junit:junit:4.12"
    const val lottie = "com.airbnb.android:lottie:2.5.0"
    const val autoValueProcessor = "com.google.auto.value:auto-value:1.4.1"
    const val kotlinStandardLibraryJDK7 = "org.jetbrains.kotlin:kotlin-stdlib-jdk7:${ProjectVersions.kotlinVersion}"
    const val dagger = "com.google.dagger:dagger:2.11"
    const val daggerCompiler = "com.google.dagger:dagger-compiler:2.11"
    const val rxJava = "io.reactivex.rxjava2:rxjava:2.+"
    const val rxAndroid = "io.reactivex.rxjava2:rxandroid:2.+"
    const val androidLifecycleExtensions = "androidx.lifecycle:lifecycle-extensions:2.0.0"
    const val androidLifecycleAnnotationProcessor = "androidx.lifecycle:lifecycle-compiler:2.0.0"
    const val dataBindingAnnotationProcessor = "com.android.databinding:compiler:3.1.0"
    const val butterknife = "com.jakewharton:butterknife:8.1.0"
    const val butterknifeCompiler = "com.jakewharton:butterknife-compiler:8.1.0"
    const val leakCanary = "com.squareup.leakcanary:leakcanary-android:1.5.4"
    const val leakCanaryNoOp = "com.squareup.leakcanary:leakcanary-android-no-op:1.5.4"
    const val processPhoenixx = "com.jakewharton:process-phoenix:2.0.0"
    const val branch = "io.branch.sdk.android:library:2.11.1"
    const val uxCam = "com.uxcam:uxcam:2.6.2@aar"
    const val onRadar = "com.onradar:sdk:1.2.19"
    const val calligraphy = "io.github.inflationx:calligraphy3:3.1.1"
    const val viewpump = "io.github.inflationx:viewpump:2.0.3"
    const val fancyButtons = "com.github.medyo:fancybuttons:1.5@aar"
    const val gifDrawable = "pl.droidsonroids.gif:android-gif-drawable:1.2.6"
    const val picassoTransformations = "jp.wasabeef:picasso-transformations:2.1.0"
    const val percentDimensions = "androidx.percentlayout:percentlayout:${ProjectVersions.androidSupportLibrary}"
    const val glide = "com.github.bumptech.glide:glide:3.7.0"
    const val realmAdapter = "io.realm:android-adapters:2.0.0"
    const val swipeLayout = "com.daimajia.swipelayout:library:1.2.0@aar"
    const val roundedImageView = "com.makeramen:roundedimageview:2.3.0"
    const val circleImageView = "com.mikhaellopez:circularimageview:4.2.0"
    const val materialMenu = "com.balysv.materialmenu:material-menu-toolbar:1.5.4"
    const val espressoCore = "androidx.test.espresso:espresso-core:3.1.0"
    const val mpAndroidChart = "com.github.PhilJay:MPAndroidChart:v3.1.0"
    const val androidxTestRunner = "androidx.test:runner:1.2.0"
    const val androidxTestRules = "androidx.test:rules:1.2.0"
    const val androidxTestCore = "androidx.test:core:1.2.0"
    const val androidxTestKtx = "androidx.test:core-ktx:1.2.0"
    const val androidxJunit = "androidx.test.ext:junit:1.1.1"
    const val androidxJunitKtx = "androidx.test.ext:junit-ktx:1.1.1"
    const val materialDialogLib = "com.shreyaspatil:MaterialDialog:2.1"
}
```
Thanks for the tip with transitive dependencies!

Yes, the `AGP version` is `3.2.0`. I will try to look into the above mentioned [Stackoverflow issue](https://stackoverflow.com/questions/58369285/illegal-class-file-class-module-info-is-missing-a-super-type-class-file-versio) and also try to upgrade the `AGP version`(although it might take sometime).

**Comment by hs-kdhillon** [comment]
Upgrading `AGP version` to `3.6.0` and `Gradle version` to `5.6.4` leads to this error:
```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':common:kaptGenerateStubsDebugKotlin'.
> Could not resolve all files for configuration ':common:kapt'.
   > Failed to transform artifact 'parceler.jar (org.parceler:parceler:1.1.5)' to match attributes {artifactType=processed-jar, org.gradle.libraryelements=jar, org.gradle.usage=java-runtime}.
      > Execution failed for JetifyTransform: /Users/karandhillon/.gradle/caches/modules-2/files-2.1/org.parceler/parceler/1.1.5/d6f8c6fcdbebda5a33fd3fe3aabf13bef67a9127/parceler-1.1.5.jar.
         > Failed to transform '/Users/karandhillon/.gradle/caches/modules-2/files-2.1/org.parceler/parceler/1.1.5/d6f8c6fcdbebda5a33fd3fe3aabf13bef67a9127/parceler-1.1.5.jar' using Jetifier. Reason: The given artifact contains a string literal with a package reference 'android.support.v4.widget' that cannot be safely rewritten. Libraries using reflection such as annotation processors need to be updated manually to add support for androidx.. (Run with --stacktrace for more details.)

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 3m 25s
34 actionable tasks: 34 executed
```

**Comment by xgouchet** [comment]
Hi @hs-kdhillon , from the error message in your comment: 

> Failed to transform '…/parceler-1.1.5.jar' using Jetifier.
> Reason: The given artifact contains a string literal with a package reference 'android.support.v4.widget' that cannot be safely rewritten. Libraries using reflection such as annotation processors need to be updated manually to add support for androidx.


 it seems one of your dependency, the [Parceler](https://github.com/johncarl81/parceler)  library is not compatible with Jetpack as it uses reflection expecting the legacy AppCompat dependencies. This issue has [already been raised on their repository](https://github.com/johncarl81/parceler/issues/362)  and it should be fixed in version Parceler version `1.1.12` or higher.

**Comment by hs-kdhillon** [comment]
Yes look likes `1.1.12-SNAPSHOT` fixes this issue. I will bring my dependency to this version and report if the issue will be fixed or not soon.

**Comment by hs-kdhillon** [comment]
Updating `Parceler` dependency resolved the issue.

**Comment by hs-kdhillon** [comment]
TLDR: Steps to resolve

1. Updated AGP version from `3.2.0` to `3.6.0` and Gradle version from `4.9` to `5.6.4`.
2. Updating `Parceler` dependency `org.parceler:parceler` from `1.1.5` to `1.1.13` and `org.parceler:parceler-api` from `.1.1.5` to `1.1.13`.

Thank you @xgouchet for your help 👍 ! Closing the issue now.
